### ISSUE TITLE
Update to ecto 3.0

### ISSUE THREAD

**Issue by scsinke** [koudelka/honeydew#64]
Is there a forecast to when ecto will be updated to 3.0?

**Comment by koudelka** [comment]
there's no timeline yet, you're more than welcome to submit a PR though. :)

**Comment by scsinke** [comment]
I tried upgrading the dependencies. I'm running in some issues. 

First:  The `simple global queue hammer async/3` and `simple global queue yield/2` are failing in master. 
```
** (exit) :not_alive
     stacktrace:
       (stdlib) slave.erl:198: :slave.start/5
       (honeydew) test/support/cluster.ex:16: Honeydew.Support.Cluster.spawn_node/1
       (honeydew) test/support/cluster_setups.ex:22: Honeydew.Support.ClusterSetups.start_node/3
       test/honeydew/global_test.exs:9: Honeydew.GlobalTest.__ex_unit_setup_1/1
       test/honeydew/global_test.exs:1: Honeydew.GlobalTest.__ex_unit__/2
```

Second: Do i need to run `cockroach` and `postgres` at the same time to let all tests run?

**Comment by scsinke** [comment]
I'm running `Elixir 1.7.4` Not sure why the other tests are failing. 

I will check tomorrow if i can run the tests with `cockroach` and `postgres` running.

**Comment by scsinke** [comment]
`Cockroach` is preventing the upgrade to `Ecto 3.0`. Ecto Poll Queue uses `db_connection ~> 1.1` for the `cockroach` connection where as from `ecto_sql ~> 3.0` `db_connection ~> 2.0` is required. The Library itself is not using postgres, so depending on if you want to adjust the examples you can upgrade.

**Comment by koudelka** [comment]
i haven't had time to look at this yet, so i don't know what's necessary to upgrade, but you might start by overriding the deps in your parent project and seeing if it Just Works.

**Comment by scsinke** [comment]
@jacobwarren I did not found a workaround yet. It has to do with `mix`. If you remove everything about `cockroach` it should works fine. But when you have `cockroach` added mix checks the dependencies for it when you add it to your project. It detects that you have a different version installed than the one specified in the `examples` folder i believe. I do have some possible options to resolve the problem. But i did not have time for it to try them.

- Make a fix for `postgrex_cdb`
- Remove examples from download via the `package manager`
- Remove support for `cockroach`

**Comment by koudelka** [comment]
@scsinke @jacobwarren i've pushed a branch with dual ecto 2.0 and 3.0 support, give it a spin.

https://github.com/koudelka/honeydew/tree/ecto-3.0

let me know how it goes, and i'll publish a new package version with it

**Comment by WernerBuchert** [comment]
I don't know if it is important, but i use the phoenixframework with Ecto3.0 and the brunch compiles fine, but i only use the in-memory-queue, but thanks for this.

**Comment by koudelka** [comment]
published version 1.3.0 to hex with ecto 3 support

https://hex.pm/packages/honeydew/1.3.0
